extensions/wikia/Discussions/controllers/EnableDiscussionsController.class.php

```diff
@@ -8,24 +8,36 @@ class EnableDiscussionsController extends \WikiaController {
 
 	public function init() {
 		$this->assertCanAccessController();
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
 	}
 
 	public function activateDiscussions() {
-		$cityId = $this->request->getInt( self::SITE_ID );
-		if ( empty( $cityId ) ) {
-			throw new BadRequestException();
-		}
+		try {
+			$cityId = $this->request->getInt( self::SITE_ID );
+			if ( empty( $cityId ) ) {
+				throw new BadRequestException();
+			}
 
-		$wiki = WikiFactory::getWikiByID( $cityId );
-		if ( empty( $wiki ) ) {
-			throw new NotFoundException();
-		}
+			$wiki = WikiFactory::getWikiByID( $cityId );
+			if ( empty( $wiki ) ) {
+				throw new NotFoundException();
+			}
 
-		( new \DiscussionsActivator( $wiki->city_id, $wiki->city_title,
-			$wiki->city_lang ) )->activateDiscussions();
+			( new \DiscussionsActivator( $wiki->city_id, $wiki->city_title,
+				$wiki->city_lang ) )->activateDiscussions();
 
-		if ( $this->request->getBool( self::CREATE_WELCOME_POST, false ) ) {
-			( new \StaffWelcomePoster() )->postMessage( $wiki->city_id, $wiki->city_lang );
+			if ( $this->request->getBool( self::CREATE_WELCOME_POST, false ) ) {
+				( new \StaffWelcomePoster() )->postMessage( $wiki->city_id, $wiki->city_lang );
+			}
+		} catch ( WikiaBaseException $exception ) {
```

Comment:
what method will cause such exception? is it necessary to try-catch all code?

Reply:
`DiscussionsActivator::activateDiscussions` throws `WikiaException` and the exceptions thrown above extend `WikiaHttpException`. Both classes extend `WikiaBaseException`.

I explained the reason to try-catch everything in https://github.com/Wikia/app/pull/15422/commits/231f607c0ebe0614db48ce1bb81974215c5acbf2